cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -874,10 +878,11 @@ func createClientBuilders(logger klog.Logger, c *config.CompletedConfig) (client
 		ClientConfig: c.Kubeconfig,
 	}
 	if c.ComponentConfig.KubeCloudShared.UseServiceAccountCredentials {
-		if len(c.ComponentConfig.SAController.ServiceAccountKeyFile) == 0 {
-			// It's possible another controller process is creating the tokens for us.
+		if len(c.ComponentConfig.SAController.ServiceAccountKeyFile) > 0 {
+			// Legacy secret-based tokens are enabled if --service-account-private-key-file is set
+			// Default behavior is that another controller process is creating the tokens for us
 			// If one isn't, we'll timeout and exit when our client builder is unable to create the tokens.
-			logger.Info("Warning: --use-service-account-credentials was specified without providing a --service-account-private-key-file")
+			logger.Info("Warning: legacy secret-based token generator is enabled (to disable unset --service-account-private-key-file)")
```

Comment:
Hi, I keep the same format of the previous warning. I believe this should be a warning so that the user is aware that a legacy controller will be enabled.

Reply:
ok, will remove the whole block.